src/main.rs

```diff
@@ -495,3 +512,38 @@ fn main_recv(addr: &str, n_conn: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::{thread, time::Duration};
+
+    #[test]
+    fn test_accepts_valid_protocol() {
+        thread::spawn(|| {
+            std::fs::File::create("a-file").unwrap();
+            main_send("127.0.0.1:1234", &["a-file".into()], 1).unwrap();
```

Comment:
Hard-coding ports is not great, but exposing the internal `TcpListener` (to be able to bind to any free port) would be a large-ish change.